wolfcrypt/src/ed25519.c

```diff
@@ -53,17 +53,28 @@
 #endif
 
 #if defined(HAVE_ED25519_SIGN) || defined(HAVE_ED25519_VERIFY)
-#define ED25519CTX_SIZE    32
+    /* set a static message string for Sig No Collisions Message SNC */
+    #define ED25519CTX_SNC_MESSAGE "SigEd25519 no Ed25519 collisions"
```

Comment:
What is the point of this refactor? These are specification fixed values. What warning was this appeasing?

Reply:
I do not recall the specific circumstances that brought this to my attention.

Having `#define ED25519CTX_SIZE  32` is not very intuitive that it is the length of the `ED25519CTX_SNC_MESSAGE` string.

Although the new code is more verbose, it also better illustrates the purpose of the length and reduces the quantity of _magic numbers_, even if it is anticipated that the string will not change.

Reply:
My preference is to revert this code and just add a code comment. This logic isn't much better. Subtracting one for the null then adding it back in. The 32 size here is from a spec.

Reply:
I removed the compile-time string length calculation and added comments as to why the specific string and length are hard-coded. 

For future readers: see [RFC8032](https://datatracker.ietf.org/doc/rfc8032/).